cmd/completion/main.go

```diff
@@ -57,56 +63,112 @@ func init() {
 func main() {
 	flag.Parse()
 
-	if hasCosign() || notaryV1URL != "" {
-		err := signImage()
-		if err != nil {
-			log.Fatal(err)
-		}
+	var report platform.ExportReport
+	_, err := toml.DecodeFile(reportFilePath, &report)
+	if err != nil {
+		log.Fatal(err, "toml decode")
 	}
 
-	logger.Println("Build successful")
-}
+	if len(report.Image.Tags) == 0 {
+		log.Fatal(errors.New("no image found in report"))
+	}
 
-func signImage() error {
-	var report platform.ExportReport
-	_, err := toml.DecodeFile(reportFilePath, &report)
+	builtImageRef := fmt.Sprintf("%s@%s", report.Image.Tags[0], report.Image.Digest)
+
+	k8sNodeKeychain, err := k8schain.NewNoClient(context.Background())
 	if err != nil {
-		return errors.Wrap(err, "toml decode")
+		log.Fatal(err)
 	}
 
-	creds, err := dockercreds.ParseMountedAnnotatedSecrets(registrySecretsDir, dockerCredentials)
+	creds, err := dockercreds.ParseBasicAuthSecrets(registrySecretsDir, dockerCredentials)
 	if err != nil {
-		return err
+		log.Fatal(err)
 	}
 
 	for _, c := range append(dockerCfgCredentials, dockerConfigCredentials...) {
 		credPath := filepath.Join(registrySecretsDir, c)
 
-		dockerCfgCreds, err := dockercreds.ParseDockerPullSecrets(credPath)
+		dockerConfigCreds, err := dockercreds.ParseDockerConfigSecret(credPath)
 		if err != nil {
-			return err
-		}
-
-		for domain := range dockerCfgCreds {
-			logger.Printf("Loading secret for %q from secret %q at location %q", domain, c, credPath)
+			log.Fatal(err)
 		}
 
-		creds, err = creds.Append(dockerCfgCreds)
+		creds, err = creds.Append(dockerConfigCreds)
 		if err != nil {
-			return err
+			log.Fatal(err)
 		}
+	}
+
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		log.Fatal(errors.Wrapf(err, "error obtaining home directory"))
+	}
+
+	err = creds.Save(filepath.Join(homeDir, ".docker", "config.json"))
+	if err != nil {
+		log.Fatal(errors.Wrapf(err, "error writing docker creds"))
+	}
+
+	keychain := authn.NewMultiKeychain(k8sNodeKeychain, creds)
+
+	metadataRetriever := cnb.RemoteMetadataRetriever{
+		Keychain:     keychain,
+		ImageFetcher: &registry.Client{},
+	}
 
-		homeDir, err := os.UserHomeDir()
+	cacheImageRef := ""
+	if cacheTag != "" {
+		cacheImageRef, err = metadataRetriever.GetCacheImage(cacheTag)
 		if err != nil {
-			return errors.Wrapf(err, "error obtaining home directory")
+			log.Fatal(err)
```

Comment:
While we are here. Could we make this non-fatal? To protect from this:https://github.com/buildpacks/lifecycle/issues/830?

Reply:
Actually, if we made this nonfatal we could have it always check for a cache image and perhaps simplify the build pod generation.

Reply:
I made a pass at improving this but maybe we can do better